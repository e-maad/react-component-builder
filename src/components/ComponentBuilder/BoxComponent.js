import React from 'react';
import ChildrenComponent from './ChildrenComponent';

const BoxComponent = ({ borderSize, children, id }) => {
    return <div id={id} className='box-component' style={{ borderWidth: borderSize }}>
        {children ? <ChildrenComponent children={children} /> : ''}
    </div>
}

export default BoxComponent;