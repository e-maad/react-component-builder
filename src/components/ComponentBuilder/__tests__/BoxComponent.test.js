import React from 'react';
import BoxComponent from '../BoxComponent';


const wrapper = shallow(<BoxComponent />);
describe('Box Component', () => {

  it('match the snapshot', () => {
    const wrapper = renderer.create(<BoxComponent />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('base element should be present', () => {
    expect(wrapper.find('.box-component').length).toBe(1);
  });
});