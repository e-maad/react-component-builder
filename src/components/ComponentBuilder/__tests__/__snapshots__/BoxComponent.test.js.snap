// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Component match the snapshot 1`] = `
<div
  className="box-component"
  style={
    Object {
      "borderWidth": undefined,
    }
  }
>
  
</div>
`;
